product/roundhouse/infrastructure.app/ConfigurationPropertyHolder.cs

```diff
@@ -40,7 +40,9 @@ public interface ConfigurationPropertyHolder
         string VersionTableName { get; set; }
         string ScriptsRunTableName { get; set; }
         string ScriptsRunErrorsTableName { get; set; }
-        string EnvironmentNames { get; set; }
+        [Obsolete("Use EnvironmentNames")]
+        string EnvironmentName { get; set; }
+        IList<string> EnvironmentNames { get; }

```

Comment:
As this is part of the API now, what may this possibly break?

Reply:
The change from a string to a list, that is.

Reply:
Also, there may be an issue with .NET 4.x where you need to also have the set here.

Reply:
I don't think there's an issue with only having the get in the interface (implementer has a private set). That should be a C# thing, not a .NET thing, and we're back on C# 5 and it works like a champ. I seem to recall doing this while the earth was still cooling, so I don't think it's a new feature.

Can you elaborate on your concern?

Reply:
Sure can - https://github.com/chocolatey/choco/blob/eaedc2b50b8573c8fd269e56921e92e26db93caf/src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs#L229.

http://stackoverflow.com/a/23809226/18475

If we want to reflect over the configuration information and output that for the logs, for the users, and we support .NET 4.0+, on .NET 4.0 it will cause errors.

Reply:
I notice that configuration class is marked [serializable] and that serialization was referenced in the linked S.O. question. Are you sure that wasn't where the issue cropped up? I have definitely had  issues with privileged sets when dealing with serialization.

I'd rather leave it private for now and open it up if we find there's a use case that we can't accommodate with the privileged set.  Adding access to a set isn't going to break anyone.

Reply:
OK, @ferventcoder , where are we going with this? I'd like to leave the code as is, but will tolerate making the set public. More than anything, I want to get this merged so we can move on.

Reply:
Merge away. If you are ready, I would say consider me more an advisor at this point, I'm going to be heads down on an important project (it does involve some RH), so you will see me pop in every once in awhile. 

This project will also be working on Cake as well, so you will see changes that will come back to UC as well.

Reply:
Cool. Didn't want to stomp on an active issue. 

@erikbra if you would do the honors, that would be great.